packages/shared/src/components/modals/SteppedModal.tsx

```diff
@@ -0,0 +1,116 @@
+import classNames from 'classnames';
+import React, {
+  Children,
+  KeyboardEvent,
+  MouseEvent,
+  ReactElement,
+  useState,
+} from 'react';
+import { Button } from '../buttons/Button';
+import TabContainer, { Tab } from '../tabs/TabContainer';
+import { ModalProps, StyledModal } from './StyledModal';
+
+interface SteppedModalProps extends ModalProps {
+  isLastStepLogin?: boolean;
+  onStepChange?: (stepBefore: number, stepNow: number) => void | Promise<void>;
+  onFinish?: () => void | Promise<void>;
+}
+
+function SteppedModal({
+  children,
+  isLastStepLogin,
+  onStepChange,
+  onFinish,
+  onRequestClose,
+  ...props
+}: SteppedModalProps): ReactElement {
+  const [step, setStep] = useState(0);
+  const getWidth = () => {
+    const count = Children.count(children);
+    return (step / (count - 1)) * 100;
+  };
+  const getBackwardsLabel = () => {
+    if (step === 0) {
+      return 'Skip';
+    }
+
+    return step === 1 ? 'Close' : 'Back';
+  };
+
+  const onBack = async (e: MouseEvent | KeyboardEvent) => {
+    if (step === 0 || step === 1) {
```

Comment:
Remember steps can be interchangeable later on

Reply:
Yep, this will be agnostic of the context of the content, the exact ordering of the children is where it will be based.